scipy/spatial/tests/test_distance.py

```diff
@@ -317,6 +317,22 @@ def test_cdist_dtype_equivalence(self):
                     for new_type in test[1]:
                         y2 = cdist(new_type(X1), new_type(X2), metric=metric)
                         _assert_within_tol(y1, y2, eps, verbose > 2)
+                        
+    def test_cdist_out(self):
+        esp = 1e-07
+        out1 = np.empty((10, 20), dtype=np.double)
+        out2 = np.empty((9, 19), dtype=np.double)
+        out3 = np.empty((100, 200), dtype=np.double)[::10, ::10]
+        out4 = np.empty((10, 20), dtype=np.int64)
+        X1 = eo['cdist-X1']
+        X2 = eo['cdist-X2']
+        Y1 = cdist(X1, X2, u('euclidean'), out=out1)
```

Comment:
Here you are testing just the euclidean metric. We could check instead against all metrics by adding a for cycle like `for metric in _METRICS_NAMES:`